### PR TITLE
ecr-viewer: show view more button when the value is long

### LINKED ISSUE #1467: Users click "View More" to see content for fields over 500 characters
This is a placeholder ticket for now, needs to be filled out in more detail, but basic details are below:

- For any fields that come in through the eCR Viewer, if there are over 500 characters, the field would be cut off at 300 characters and users would have to click a "View More" button to see the full content
- Cutting at exactly 300 characters
- Handle this in the data display so it applies everywhere
- They can collapse the content again with a "View Less" button
- Clickable area should just be the View More/Less button, but Figma won't let me do that easily

[Figma Designs](https://www.figma.com/file/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?type=design&node-id=1957%3A26202&mode=design&t=MTFSUbcZ86xrQf61-1)
[Clickable Figma Prototype](https://www.figma.com/proto/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?page-id=1957%3A26202&type=design&node-id=1957-26315&viewport=220%2C2071%2C0.3&t=3F6hLaLUhcAbjQ9c-1&scaling=min-zoom&starting-point-node-id=1957%3A26315&mode=design)

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- When a field value is longer than 500 characters, only show the first 300 characters and a view more button.
- Create a recursive method to determine the character length of an [ReactNode](https://reactnative.dev/docs/react-node)
- Create a recursive method to cut at 300 characters. 
- When this PR goes in - I will move these functions out of utils and into it's own file.

## Related Issue
Fixes #1467 

## Additional Information
Can be viewed and tested locally:
- http://localhost:3000/view-data?id=1.2.840.114350.1.13.202.3.7.8.688883.486#clinical-notes
- http://localhost:3000/view-data?id=1.2.840.114350.1.13.478.3.7.8.688883.230886#lab-results-from-vumc-cerner-lab

## Visual Review
Video:
https://github.com/CDCgov/phdi/assets/10108172/5bf63de0-0e1d-4e35-b440-d315e6c82084
Screenshots:
image
image

image
image

### CODE REVIEW

**Comment by BobanL** [comment]
@sarahtress - ready for review! Screenshots & video in the body of this PR!

**Comment by sarahtress** [comment]
@BobanL this looks great, thanks!

File: containers/ecr-viewer/src/app/utils.tsx [hunk 1]
```diff
@@ -1158,14 +1157,142 @@ export const DataDisplay: React.FC<{
...
+  const valueLength = getReactNodeLength(children);
+  const hiddenValue = viewMoreElement(children, cutLength, setHidden).value;
+  useEffect(() => {
+    if (valueLength > maxLength) {
+      if (hidden) {
+        setFieldValue(hiddenValue);
+      } else {
+        setFieldValue(
+          <>
+            {children}{" "}
```
Comment (KennethSkylight) [inline containers/ecr-viewer/src/app/utils.tsx:1190]: What does this space do?

File: containers/ecr-viewer/src/app/utils.tsx [hunk 2]
```diff
@@ -1158,14 +1157,142 @@ export const DataDisplay: React.FC<{
...
+};
+
+/**
+ * Create an element with `remainingLength` length followed by a view more button
+ * @param value - the value that will be cut
+ * @param remainingLength - the length of how long the returned element will be
+ * @param setHidden - a function used to signify that the view more button has been clicked.
+ * @returns - an object with the shortened value and the length left over.
+ */
+const viewMoreElement = (
```
Comment (KennethSkylight) [inline containers/ecr-viewer/src/app/utils.tsx:1232]: Needs clarification on name

File: containers/ecr-viewer/src/app/utils.tsx (hunk 1 above)
Comment (KennethSkylight) [reply containers/ecr-viewer/src/app/utils.tsx:1190]: You could use &nbsp; instead

**Review by KennethSkylight — COMMENTED** [review]
Just a few things

File: containers/ecr-viewer/src/app/utils.tsx (hunk 1 above)
Comment (BobanL) [reply containers/ecr-viewer/src/app/utils.tsx]: `&nbsp;` just gets converted to space in comments 😆

File: containers/ecr-viewer/src/app/utils.tsx (hunk 2 above)
Comment (BobanL) [reply containers/ecr-viewer/src/app/utils.tsx]: I renamed it to `trimField` - hopefully it makes it more clear.

**Review by KennethSkylight — APPROVED** [review]

**Pull request merged**